nym-vpn-core/crates/nym-vpn-account-controller/src/util.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright 2024 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-3.0-only
+
+use std::{fs, path::PathBuf};
+
+use crate::Error;
+
+// TODO: implement functionality where the owning code of these files delete them instead. To
+// protect us against the names drifting out of sync.
+
+pub fn remove_files_for_account(data_dir: PathBuf) -> Result<(), Error> {
+    // Delete all files in data_dir that matches any of these patterns:
+    // credentials_database.db
+    // *.sqlite
+    // *.pem
+
+    for file in fs::read_dir(data_dir).map_err(Error::internal)? {
+        let file = match file {
+            Ok(file) => file,
+            Err(err) => {
+                tracing::error!("failed to read file in data directory: {:?}", err);
+                break;
```

Comment:
`break` will abort dir file iteration
I think it's better to just ignore any no matching files (like directory) and continue to iterate through

Reply:
ah yes good catch, I mean `continue`

Reply:
also can change the `error` lvl to `warn` imo